plugins/logging/file

```diff
@@ -279,13 +279,14 @@ sub hook_logging {
     if (   !$self->{_f}
         || !$self->{_nosplit}
         || !$transaction
+        || !UNIVERSAL::can($transaction, 'notes')
         || !$transaction->notes('file-logged-this-session'))
     {
         unless (defined $self->maybe_reopen($transaction)) {
             return DECLINED;
         }
-        if (UNIVERSAL::can($transaction,'isa')) {
-            $transaction->notes('file-logged-this-session', 1) if $transaction;
+        if (UNIVERSAL::can($transaction,'isa') && $transaction && UNIVERSAL::can($transaction, 'notes')) {
```

Comment:
You're raising some of the same questions that I raised in the original post.
I made the minimal change to fix the issue that I was having, because I don't understand all the subtleties of the code, but I'll be glad to post an update when we have answers...